### ISSUE TITLE
Convert every component into its own module (sugestion).

### ISSUE THREAD

**Issue by mishelashala** [lambdabricks/bricks-front-react#5]
For example, if you have the component `Library` create a directory called `library`and inside sub folders (containers, components, propTypes, actions, etc) and expose them via an API.

``` javascript
// library/index.js
const LibraryContainer = require('./containers/library');
const Library = require('./components/library');
const PropTypes = require('./propTypes');

module.exports = {
  component: Library,
  container: LibraryContainer,
  propTypes: PropTypes
};
```

``` javascript
// library/components/library.js
...
module.exports = function (props) {
  // ...
}
```

``` javascript
// library/containers/library.js
...
module.exports = connect(...)(Library);
```

**Comment by pel-daniel** [comment]
I thought about doing something similar at the start of the project but never get to it. I was thinking of something similar to what is called [pods in ember](https://ember-cli.com/user-guide/#pod-structure). 

Do you have some links of projects already organized this way ? Or discussion on pros and cons of each mode ?

**Comment by mishelashala** [comment]
This is for Redux apps, but works to react (stand-alone) too.
http://jaysoo.ca/2016/02/28/organizing-redux-application/

**Comment by pel-daniel** [comment]
Thanks for the link, the organization proposed is interesting.

I would like to ask you if you found the code of this project hard to follow, or have any thoughts on specific ways this project would benefit from the reorganization.

**Comment by pel-daniel** [comment]
@mishelashala I just found this link, it looks really cool!! [http://antontelesh.github.io/architecture/2016/03/16/fractal-architecture.html](http://antontelesh.github.io/architecture/2016/03/16/fractal-architecture.html)
